### DISCUSSION TITLE
Dart package fvm is failing to install any version Exception: Git clone failed

### DISCUSSION THREAD

**Discussion by muhammadmubashir24** [leoafarias/fvm#687] [Q&A] [answered]
![image](https://github.com/leoafarias/fvm/assets/132802289/e82d6365-7bdd-48a5-b658-263d1f4c95c8)

i need help if i run this commands "fvm install v1.2.1" its say error git clone faileed

**Comment by muhammadmubashir24** [comment]
> ![image](https://private-user-images.githubusercontent.com/132802289/311408400-e82d6365-7bdd-48a5-b658-263d1f4c95c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5NTY5ODEsIm5iZiI6MTcwOTk1NjY4MSwicGF0aCI6Ii8xMzI4MDIyODkvMzExNDA4NDAwLWU4MmQ2MzY1LTdiZGQtNDhhNS1iNjU4LTI2M2QxZjRjOTVjOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwOVQwMzU4MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02YzBhYTE2YzYyNzBhMzI4ZWY5MGYwODRlNzBiZWIxNDZjZjA1MTFiZTZhZmZlZGJjYWYzYTY0NTU0MjAyOTNjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.sd03mUni23VGDgRzaCl8tRzI_iFgiMvdIgGnHji0xZk)
> 
> i need help if i run this commands "fvm install v1.2.1" its say error git clone faileed

i found another error fatal: 'origin' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
✗ Failed to install 3.19.3 (235.9s)
✗ Failed to ensure 3.19.3 is cached.

ProcessException: Cloning into 'C:\Users\muham\fvm\versions\3.19.3'...
remote: Enumerating objects: 500143, done.
remote: Counting objects: 100% (552/552), done.
remote: Compressing objects: 100% (341/341), done.
error: 4681 bytes of body are still expected6 MiB | 79.00 KiB/s
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
  Command: git clone --progress -c advice.detachedHead=false -b stable --reference C:\Users\muham\fvm\cache.git https://github.com/flutter/flutter.git C:\Users\muham\fvm\versions\3.19.3

**Reply by leoafarias** [reply]
@muhammadmubashir24, when installing this version, does a plain git clone work outside of FVM?

**Reply by muhammadmubashir24** [reply]
I dnt know how to check?

**Reply by leoafarias** [reply]
```
git clone --progress -c advice.detachedHead=false -b stable --reference C:\Users\muham\fvm\cache.git https://github.com/flutter/flutter.git C:\Users\muham\fvm\versions\3.19.3
```

If it does work try running the command again through `fvm`

**Reply by muhammadmubashir24** [reply]
fvm use v1.2.1      
┌────────────────────────────────────────────────────────────────────────────┐
│ ⚠ Version mismatch detected: cache version is 3.19.3, but expected v1.2.1. │
└────────────────────────────────────────────────────────────────────────────┘
This can occur if you manually run "flutter upgrade" on a cached SDK.

✔ How would you like to resolve this? · Move 3.19.3 to the correct cache directory and reinstall v1.2.1
Moving SDK to the correct cache directory...
Removing incorrect SDK version...
fatal: 'origin' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.                                                                                                                                         
✓ Flutter SDK: SDK Version : v1.2.1 installed! (146.1s)                                                                                                            Setting up Flutter SDK: v1.2.1                                                                                                                                                                                                                                                                                                        Checking Dart SDK version...                                                                                                                                       Downloading Dart SDK from Flutter engine 3757390fa4b00d2d261bfdf5182d2e87c9113ff9...                                                                               
Unzipping Dart SDK...                                                                                                                                              
Building flutter tool...
Running pub upgrade...
  ╔════════════════════════════════════════════════════════════════════════════╗
  ║ A new version of Flutter is available!                                     ║
  ║                                                                            ║
  ║ To update to the latest version, run "flutter upgrade".                    ║
  ╚════════════════════════════════════════════════════════════════════════════╝


Flutter 1.2.1 • channel dev • https://github.com/flutter/flutter.git
Framework • revision 8661d8aecd (5 years ago) • 2019-02-14 19:19:53 -0800
Engine • revision 3757390fa4
Tools • Dart 2.1.2 (build 2.1.2-dev.0.0 0a7dcf17eb)

✓ Flutter SDK: SDK Version : v1.2.1 is setup
⠋ Resolving dependencies...... (1815.2s)
its still running resolving dependencies how much take take ? this query

**Reply by muhammadmubashir24** [reply]
its showing Resolving dependencies...... (4272.4s) still running how much take this time almost waiting 42min

**Reply by leoafarias** [reply]
hmmm... ok that might be another issue, but it seems that its completed successfully

**Reply by muhammadmubashir24** [reply]
ok

**Reply by muhammadmubashir24** [reply]
![image](https://github.com/leoafarias/fvm/assets/132802289/73c4309d-5e96-4c09-91fd-7271ffd32082)
i am facing this issue if i run fvm flutter run

and fvm flutter doctor showing
![image](https://github.com/leoafarias/fvm/assets/132802289/d181c62f-9f31-4f00-a9a1-750807a1dd52)

but andriod studio install i run flutter doctor without fvm its all ok what can i do
![image](https://github.com/leoafarias/fvm/assets/132802289/6fc4d403-bc22-4d34-8c00-f473b35a4bfd)

**Reply by leoafarias** [reply]
@muhammadmubashir24, are you running `fvm flutter run` within the Android Studio terminal? Could you run it similarly to how you did with `flutter doctor`, and vice versa, to check for any environment issues?

**Reply by muhammadmubashir24** [reply]
yes if i run in specific projec within the android studio terminal its showing above error i already attached but if i run this outside cmd terminal outside the project so no issue found showed

**Comment by burhanrashid52** [comment, marked as answer]
The way I found to fix this is to first use HTTP/1.1 and then increase the post buffer size.

Use HTTPS/1.1
```bash
git config --global http.version HTTP/1.1
```

And then increase the buffer size using
```bash
git config --global http.postBuffer 2038500000
```

However, the trick is to increase the buffer based on the bytes mentioned in the error message that says it cannot download. For example, the git error looks like this:

```bash
    Receiving objects:   5% (28560/532810), 19.44 MiB | 336.00 KiB/s
    error: RPC failed; curl 18 transfer closed with outstanding read data remaining
    error: 5319 bytes of body are still expected
    fetch-pack: unexpected disconnect while reading sideband packet
    fatal: early EOF
    fatal: fetch-pack: invalid index-pack output
```

Here, the line `error: 5319 bytes of body are still expected` mentions 5319 bytes. So, keep increasing the buffer size until it gets small enough, i.e., zero, to get this working.
```bash
git config --global http.postBuffer {additionalbytes}
```

**Reply by Riyadhalali** [reply]
thanks it worked

**Comment by leonheldattoradex** [comment]
I don't even use this project, but I had the same issue and ended up here. Posting what worked for me somewhere because it's probably useful anyway: the Starbucks WiFi was the issue.
